lua/nvim-tree/renderer/builder.lua

```diff
@@ -413,6 +413,8 @@ end
 function Builder:_build_line(node, idx, num_children, unloaded_bufnr)
   local copy_paste = require "nvim-tree.actions.fs.copy-paste"
 
+  require("nvim-tree.diagnostics").update_node_severity_level(node)
```

Comment:
This one is very concerning from a performance perspective;  we go out of our way to avoid such calls on drawing each line.

Could we improve via one of:
* do this once for all nodes prior to rendering
* short-circuit this so that it's only called when there is no diagnostic info